docs/guides/pod-readiness-gates.md

```diff
@@ -0,0 +1,144 @@
+
+# Step 3: Patch the webhook CA bundle to exactly the cert being used.
+# This will ensure Kubernetes API server is able to trust the certificate presented by the webhook.
+# This step would not be required if you are using a signed certificate that is already trusted by the API server
+CERT_B64=$(cat tls.crt | base64)
+kubectl patch mutatingwebhookconfigurations.admissionregistration.k8s.io $WEBHOOK_NAME \
+    --namespace $WEBHOOK_NAMESPACE --type='json' \
+    -p="[{'op': 'replace', 'path': '/webhooks/0/clientConfig/caBundle', 'value': '${CERT_B64}'}]"
+```
+
+## Configuration
+Pod readiness gate support is enabled by default on the AWS Gateway API controller. To enable the feature, you must apply a label to each of the namespaces you would like to use this feature. You can create and label a namespace as follows -
+
+```
+$ kubectl create namespace example-ns
+namespace/example-ns created
+
+$ kubectl label namespace example-ns aws-application-networking-k8s/pod-readiness-gate-inject=enabled
+namespace/example-ns labeled
+
+$ kubectl describe namespace example-ns
+Name:         example-ns
+Labels:       aws-application-networking-k8s/pod-readiness-gate-inject=enabled
+              kubernetes.io/metadata.name=example-ns
+Annotations:  <none>
+Status:       Active
+```
+
+Once labelled, the controller will add the pod readiness gates to all subsequently created pods.
+
+The readiness gates have the prefix `<TBD>` and the controller injects the config to the pod spec only during pod creation.
+
+## Object Selector
+The default webhook configuration matches all pods in the namespaces containing the label `aws-application-networking-k8s/pod-readiness-gate-inject=enabled`. You can modify the webhook configuration further to select specific pods from the labeled namespace by specifying the `objectSelector`. For example, in order to select ONLY pods with `aws-application-networking-k8s/pod-readiness-gate-inject: enabled` label instead of all pods in the labeled namespace, you can add the following `objectSelector` to the webhook:
+```
+  objectSelector:
+    matchLabels:
+      aws-application-networking-k8s/pod-readiness-gate-inject: enabled
+```
+To edit,
+```
+$ kubectl edit mutatingwebhookconfigurations aws-appnet-gwc-mutating-webhook
+  ...
+  name: mpod.gwc.k8s.aws
+  namespaceSelector:
+    matchExpressions:
+    - key: aws-application-networking-k8s/pod-readiness-gate-inject
+      operator: In
+      values:
+      - enabled
+  objectSelector:
+    matchLabels:
+      aws-application-networking-k8s/pod-readiness-gate-inject: enabled
+  ...
+```
+When you specify multiple selectors, pods matching all the conditions will get mutated.
+
+## Disabling the readiness gate inject
+You can specify the controller flag `--enable-pod-readiness-gate-inject=false` during controller startup to disable the controller from modifying the pod spec.
```

Comment:
Is this feature implemented? having hard time finding this

Reply:
Good catch. It is not. This is something in the load balancer controller. Will scrub through this file again now that we have the other PR open too.

Reply:
removed